stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpResource.java

```diff
@@ -233,7 +216,7 @@ private static Annotation getResourceConfigAnnotation(Resource resource, String
 
     private static void processResourceCors(HttpResource resource, HttpService service) {
         CorsHeaders corsHeaders = resource.getCorsHeaders();
-        if (!corsHeaders.isAvailable()) {
+        if (corsHeaders == null || !corsHeaders.isAvailable()) {
```

Comment:
Please check whether we need isAvailable() check anymore as now we have a separate struct for cors. Initially, this was added as a flag to find the availability of cors fields in service annotation.